server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -541,33 +541,24 @@ private boolean checkForChangedProducts(Product incomingProduct, Pool existingPo
         return productsChanged;
     }
 
-    private boolean checkForChangedDerivedProducts(Pool pool, Pool existingPool,
+    public boolean checkForChangedDerivedProducts(Pool pool, Pool existingPool,
```

Comment:
This works, but still kinda violates the separation from the original implementation. Also, why did the access change?

In the interest of time, this is kind of what I was suggesting:

```
private boolean checkForChangedDerivedProducts(Pool pool, Pool existingPool, Map<String, Product> changedProducts) {

  boolean dpChanged = false;
  Product existingDerived = existingPool.getDerivedProduct();
  Product incomingDerived = pool.getDerivedProduct();

  if (incomingDerived != null) {
    dpChanged = (existingDerived == null) ||
      (existingDerived.getId() != null && !existingDerived.getId().equals(incomingDerived.getId())) ||
      (changedProducts != null && changedProducts.containsKey(existingDerived.getId()));
  }
  else {
    dpChanged = existingDerived != null;
  }

  if (dpChanged) {
    existingPool.setDerivedProduct(incomingDerived);
  }

  return dpChanged;
}
```

This still keeps the evaluation in one block, and the action to take on that evaluation result in another.

Reply:
ahh while doing unit testing for certain scenarios I changed the method access. Corrected it with above code. Will squash commits after PR goes all green.